setup.py

```diff
@@ -68,16 +67,17 @@ def run(self):
 if any([arg in CMDS_NOCYTHONIZE for arg in sys.argv]):
     ext_modules = []
 else:
-    extension = Extension('{}._{}'.format(NAME, NAME),
-                          sources=[os.path.relpath(CYTHON_FNAME, BASEDIR)],
-                          define_macros=DEFINE_MACROS,
-                          include_dirs=[numpy.get_include(),])
-
-    ext_modules = cythonize(
-        extension,
-        compiler_directives=COMPILER_DIRECTIVES,
-        language_level=3,
-    )
+    ext_modules = [Extension('{}._{}'.format(NAME, NAME),
+                  sources=[os.path.relpath(CYTHON_FNAME, BASEDIR)],
+                  define_macros=DEFINE_MACROS,
+                  include_dirs=[numpy.get_include(),])]
+    for e in ext_modules:
+        e.cython_directives = {'language_level': "3"} 
+        e.compiler_directives = COMPILER_DIRECTIVES
+    # remove _cftime.c file if it exists, so cython will recompile _cftime.pyx.
+    if len(sys.argv) >= 2:
```

Comment:
not sure what the intention for this check is. Seems to me there are many ways to end up executing this code with completely different commands so I'm a bit unconfortable with it. Can you explain ?

Reply:
the idea is that anytime you run setup.py, you will want to recompile _cftime.pyx.  If there is a left-over _cftime.c file, then cython won't run, so this code removes it if it exists.

Reply:
From [the setuptools docs](https://setuptools.pypa.io/en/latest/userguide/ext_modules.html):

"When your [Cython](https://pypi.org/project/Cython) extension modules are declared using the [setuptools.Extension](https://setuptools.pypa.io/en/latest/userguide/ext_modules.html#setuptools.Extension) class, setuptools will detect at build time whether Cython is installed or not.

If Cython is present, then setuptools will use it to build the .pyx files. Otherwise, setuptools will try to find and compile the equivalent .c files (instead of .pyx). These files can be generated using the [cython command line tool](https://cython.readthedocs.io/en/stable/src/userguide/source_files_and_compilation.html)."

so you don't really need to use cythonize explicitly.

Reply:
> the idea is that anytime you run setup.py, you will want to recompile _cftime.pyx

but why would this be captured specifically by the number of CLI arguments ? This is what I'm having a hard time wrapping my head around.